avro-serializer/src/main/java/io/confluent/kafka/serializers/AbstractKafkaAvroDeserializer.java

```diff
@@ -252,4 +231,89 @@ private Schema getReaderSchema(Schema writerSchema) {
     }
     return readerSchema;
   }
+
+  class DeserializationContext {
+    private final String topic;
+    private final Boolean isKey;
+    private final ByteBuffer buffer;
+    private final int schemaId;
+
+    private Schema cachedSchemaFromRegistry = null;
+    private Schema cachedSchemaForDeserialize = null;
+
+    DeserializationContext(final String topic, final Boolean isKey, final byte[] payload) {
+      this.topic = topic;
+      this.isKey = isKey;
+      this.buffer = getByteBuffer(payload);
+      this.schemaId = buffer.getInt();
+    }
+
+    Schema getSchemaFromRegistry() {
```

Comment:
Perhaps call it `schemaFromRegistry()`

Reply:
```suggestion
    Schema schemaFromRegistry() {
```

Reply:
@mageshn I've added a commit to address these issues.